src/io/flutter/run/FlutterRunConfigurationProducer.java

```diff
@@ -51,6 +52,12 @@ public boolean isConfigurationFromContext(final @NotNull FlutterRunConfiguration
     return dartFile != null && dartFile.getPath().equals(configuration.getRunnerParameters().getFilePath());
   }
 
+  @Override
+  public boolean shouldReplace(ConfigurationFromContext self, ConfigurationFromContext other) {
+    // Prefer Flutter runner to plain Dart runner for Flutter code.
+    return other.getConfigurationType().getId().equals("DartCommandLineRunConfigurationType");
```

Comment:
Right. It's a hard call.  We don't have a good story for this.  Let's say a string is fine but maybe store it as a constant in the `DartPlugin` class (that's where we're trying to consolidate our plugin dependencies)?  Alternatively, add a method there?  Something like:

    static boolean isDartCommandLineRunConfig(ConfigurationFromContext config) { ... }